packages/cli/src/commands/update.js

```diff
@@ -0,0 +1,316 @@
+      } else {
+        task.title = 'No upgrade is available'
+      }
+
+      writeUpgradeFile(updateData)
+    },
+  })
+
+  try {
+    setLock(UPDATE_LOCK)
+    await updateTasks.run()
+    if (upgradeAvailable && !silent) {
+      showUpgradeAvailableMessage()
+    }
+    unsetLock(UPDATE_LOCK)
+  } catch (e) {
+    unsetLock(UPDATE_LOCK)
+    errorTelemetry(process.argv, e.message)
+    console.error(c.error(e.message))
+    process.exit(e?.exitCode || 1)
+  }
+}
+
+async function getUpdateVersionStatus() {
+  // Read package.json and extract the @redwood/core version
+  const packageJson = JSON.parse(
+    fs.readFileSync(path.join(getPaths().base, 'package.json'))
+  )
+  let localVersion = packageJson.devDependencies['@redwoodjs/core']
+
+  // Remove any leading non-digits, i.e. ^ or ~
+  while (!(localVersion.charAt(0) >= '0' && localVersion.charAt(0) <= '9')) {
+    localVersion = localVersion.substring(1)
+  }
+
+  // Determine if the user has a tag (e.g. -rc, -canary), if so extract the tag from the version
+  let tag = ''
+  if (localVersion.includes('-')) {
+    tag = localVersion.substring(localVersion.indexOf('-') + 1).trim()
+
+    // Validate the local version, just incase the user has manually fiddled with it
+    // no if needed to check as it throws an error if not successful, will be caught by listr
+    validateTag(tag)
+  }
+
+  // Fetch the latest version from npm registry
+  let remoteVersion
+  try {
+    remoteVersion = await latestVersion(
+      '@redwoodjs/core',
+      tag ? { version: tag } : {}
+    )
+  } catch (e) {
+    throw new Error('Could not find the latest version')
+  }
+
+  // Is remote version higher than local?
+  const upgradeAvailable = semver.gt(remoteVersion, localVersion)
+
+  // Don't change the skip version
+  let skipVersion = readUpgradeFile().skipVersion
+
+  // Build an object with some details to be returned. Avoids the need for more parsing or remote calls elsewhere
+  const versionsStatus = {
+    localVersion,
+    remoteVersion,
+    skipVersion,
+    upgradeAvailable,
+    lastChecked: Date.now(),
+  }
+
+  return versionsStatus
+}
+
+function getCheckPeriod() {
+  return (
+    (process.env.REDWOOD_BACKGROUND_UPDATES_CHECK_PERIOD || 24 * 60) * 60000
+  )
+}
+
+function getShowPeriod() {
+  return (process.env.REDWOOD_BACKGROUND_UPDATES_SHOW_PERIOD || 60) * 60000
+}
+
+export function isUpdateCheckDue() {
+  const updateData = readUpgradeFile()
+  return updateData.lastChecked < Date.now() - getCheckPeriod()
+}
+
+export function shouldShowUpgradeAvailableMessage() {
+  const updateData = readUpgradeFile()
+  return (
+    updateData.upgradeAvailable &&
+    !(updateData.skipVersion == updateData.remoteVersion) &&
+    updateData.lastShown < Date.now() - getShowPeriod()
+  )
+}
+
+function getUpgradeFilePath() {
+  return path.join(getPaths().base || '/tmp', '.redwood', 'update-data.json')
```

Comment:
Mind explaining the `/tmp` part? And I don't think `getPaths.base()` would every be undefined at this point, that's pretty much a fatal error and would cause an error to be thrown earlier

Reply:
Well I was about to say I found it being already used within the framework but when I went to search I only found it in my own code... Hmm 😆 

More than happy to remove the `|| /tmp`.